syntax = "proto3";

package workflow;

enum ExecutionStatus {
  STATUS_UNSPECIFIED = 0;
  PENDING = 1;
  RUNNING = 2;
  COMPLETED = 3;
  FAILED = 4;
  CANCELLED = 5; 
}

message ExecuteWorkflowRequest {
  string workflow_id = 1; 
  // map<string, string> parameters = 2; 
}

message WorkflowExecutionUpdate {
  string execution_id = 1;      
  string workflow_id = 2;      
  ExecutionStatus status = 3;  
  string current_task_id = 4;   
  string current_task_name = 5;
  string message = 6;          
  // int32 progress_percent = 7; 
}

message GetWorkflowStatusRequest {
  string execution_id = 1; // ID of the specific run to query
}

message WorkflowStatusResponse {
  string execution_id = 1;
  string workflow_id = 2;
  ExecutionStatus status = 3;
  string last_message = 4; 
  // google.protobuf.Timestamp finished_at = 5;
}

service WorkflowService {
  rpc ExecuteWorkflow(ExecuteWorkflowRequest) returns (stream WorkflowExecutionUpdate);

  rpc GetWorkflowStatus(GetWorkflowStatusRequest) returns (WorkflowStatusResponse);
}
